test/ProtocolFeesImplementation.t.sol

```diff
@@ -0,0 +1,202 @@
+    OutOfBoundsProtocolFeeControllerTest,
+    RevertingProtocolFeeControllerTest,
+    OverflowProtocolFeeControllerTest,
+    InvalidReturnSizeProtocolFeeControllerTest
+} from "../src/test/ProtocolFeeControllerTest.sol";
+
+contract ProtocolFeesTest is Test, GasSnapshot, Deployers {
+    using CurrencyLibrary for Currency;
+    using PoolIdLibrary for PoolKey;
+    using ProtocolFeeLibrary for uint24;
+
+    event ProtocolFeeControllerUpdated(address feeController);
+    event ProtocolFeeUpdated(PoolId indexed id, uint24 protocolFee);
+
+    uint24 constant MAX_PROTOCOL_FEE_BOTH_TOKENS = (1000 << 12) | 1000; // 1000 1000
+
+    ProtocolFeesImplementation protocolFees;
+
+    function setUp() public {
+        protocolFees = new ProtocolFeesImplementation(5000);
+        feeController = new ProtocolFeeControllerTest();
+        (currency0, currency1) = deployAndMint2Currencies();
+        MockERC20(Currency.unwrap(currency0)).transfer(address(protocolFees), 2 ** 255);
+    }
+
+    function test_setProtocolFeeController_succeedsNoRevert() public {
+        assertEq(address(protocolFees.protocolFeeController()), address(0));
+        vm.expectEmit(false, false, false, true, address(protocolFees));
+        emit ProtocolFeeControllerUpdated(address(feeController));
+        protocolFees.setProtocolFeeController(feeController);
+        assertEq(address(protocolFees.protocolFeeController()), address(feeController));
+    }
+
+    function test_setProtocolFeeController_revertsWithNotAuthorized() public {
+        assertEq(address(protocolFees.protocolFeeController()), address(0));
+
+        vm.prank(address(1)); // not the owner address
+        vm.expectRevert("UNAUTHORIZED");
+        protocolFees.setProtocolFeeController(feeController);
+        assertEq(address(protocolFees.protocolFeeController()), address(0));
+    }
+
+    function test_setProtocolFee_succeeds_gas() public {
+        PoolKey memory key = PoolKey(currency0, currency1, 3000, 60, IHooks(address(0)));
+        protocolFees.setProtocolFeeController(feeController);
+        protocolFees.setPrice(key, 1);
+        vm.prank(address(feeController));
+        vm.expectEmit(true, false, false, true, address(protocolFees));
+        emit ProtocolFeeUpdated(key.toId(), MAX_PROTOCOL_FEE_BOTH_TOKENS);
+        protocolFees.setProtocolFee(key, MAX_PROTOCOL_FEE_BOTH_TOKENS);
+        snapLastCall("set protocol fee");
+    }
+
+    function test_setProtocolFee_revertsWithInvalidCaller() public {
+        protocolFees.setProtocolFeeController(feeController);
+        vm.expectRevert(IProtocolFees.InvalidCaller.selector);
+        protocolFees.setProtocolFee(key, 1);
+    }
+
+    function test_setProtocolFee_revertsWithInvalidFee() public {
+        protocolFees.setProtocolFeeController(feeController);
+        vm.prank(address(feeController));
+        vm.expectRevert(IProtocolFees.InvalidProtocolFee.selector);
+        protocolFees.setProtocolFee(key, MAX_PROTOCOL_FEE_BOTH_TOKENS + 1);
```

Comment:
could you add 2 more lines to check it also reverts for `MAX_PROTOCOL_FEE_BOTH_TOKENS + (1 << 12)`